src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -204,7 +204,10 @@ internal sealed partial class ConsoleHost
                 {
                     try
                     {
-                        TaskbarJumpList.CreateElevatedEntry(ConsoleHostStrings.RunAsAdministrator);
+                        // APIs are STA only and until issue 7216 is not resolved, PowerShell operates still in MTA mode.
```

Comment:
I would remove the `Task.Run()` call entirely. This is putting additional pressure on the threadpool which contains all MTA based threads just to start a non-threadpool thread that is in an STA. The `try`/`catch` is then unnecessary and should be put inside the lambda.

``` csharp
   var thread = new Thread(() =>
   {
      try
      {
            TaskbarJumpList.CreateElevatedEntry(ConsoleHostStrings.RunAsAdministrator));
      }
      catch
      {
      }
   });
   thread.SetApartmentState(ApartmentState.STA);
   thread.Start();
```

Also note that `// APIs are STA only and until issue 7216 is not resolved, PowerShell operates still in MTA mode.` may be true but running in a `Task` hides that fact since it will consume a threadpool thread.

Reply:
Thanks, I removed the `Task.Run` call but kept the global catch as `thread.Start()` can still throw exceptions like e.g. `OutOfMemoryException` and since this part of the code is not crucial to the workings of PowerShell, it is better to just ignore any exceptions here.